Improving WebServer Example (#10111)
espressif/arduino-esp32: libraries/WebServer/examples/WebServer/README.md

> This example requires some space for a filesystem and runs fine boards with 4 MByte flash using the following options:

Reviewer: Please update also this line as suggested, so its clear that the next part is about partition tables.
```suggestion
This example requires some space for a filesystem and runs fine on supported SoCs with 4 MByte or more flash by selecting proper partition table:
```

Author: yes, better wording.Thanks

* Partition Scheme: Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS)
    but LittleFS will be used in the partition (not SPIFFS)

> or
> 
> * Board: Arduino Nano ESP32
> * Partition Scheme: "With FAT partition"
> 
> 

Reviewer: This section is not needed at all, as you can run it on any board having a ESP32.
```suggestion
```

Can you please remove it and maybe change the section above to something like this:
```
This example requires some space for a filesystem and runs fine on supported SoCs with 4 MByte or more flash by selecting proper partition table:

* For using SPIFFS(LittleFS): Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS)
* For using FATFS: Default 4MB with ffat (1.2MB APP/1.5MB FATFS)

```

Currently, this example supports the following targets.

| Supported Targets | ESP32 | ESP32-S2 | ESP32-S3 | ESP32-C3 |
| ----------------- | ----- | -------- | -------- | -------- |

> |                   | yes   | no       | yes      | yes      |

Reviewer 2: What is the limitation on ESP32-S2?

Author: It compiles and maybe works, but I since I initially wrote the example I could not test it on S2.

Reviewer: I have tested to be sure today, work fine on all chips.

> | Supported Targets | ESP32 | ESP32-S2 | ESP32-S3 | ESP32-C3 |
> | ----------------- | ----- | -------- | -------- | -------- |
> |                   | yes   | no       | yes      | yes      |

Reviewer: Please fix S2 and add a C6 to the supported targets. Just tested them and works fine :)
```suggestion
| Supported Targets | ESP32 | ESP32-S2 | ESP32-S3 | ESP32-C3 | ESP32-C6 |
| ----------------- | ----- | -------- | -------- | -------- | -------- |
|                   | yes   | yes      | yes      | yes      | yes      |
```

> #define TIMEZONE "CET-1CEST,M3.5.0,M10.5.0/3"
> ```

## Troubleshooting

Have a look in the Serial output for some additional runtime information.

## Changes

* 2024-08-02 -- Fixing for board implementation 3.0.4 ff.
* 2024-08-02 -- Support for FAT

> * 2024-08-02 -- Tested with Arduino Nano ESP32

Reviewer: This is not something for a change-log, please remove it.
```suggestion
```
